src/extension_commands.ml

```diff
@@ -841,7 +866,9 @@ module MerlinJump = struct
       QuickPick.onDidHide
         quickPick
         ~listener:(fun () ->
-          if !selected_item then () else show_selection initial_selection text_editor;
+          if !selected_item
+          then ()
+          else Disposable.dispose (move_cursor_after_quickpick_close initial_selection ());
```

Comment:
I think this does nothing: you start listening to an event and then dispose of it immediately.

I'm not sure of the best way to do it myself, but maybe we should, when the quickpick is being shown, watch for such events so that we set a flag when the user changed the selection to handle the onDidHide event accordingly.